[ABI] Added signedAsData bit to method descriptor flags.

nate-chandler opened the pull request:
It is necessary to determine whether a given method in a vtable or a
witness table is async in order to decide whether to auth/sign as data
or as code.

nate-chandler commented:
@swift-ci please smoke test

aschwaighofer reviewed (approved):
Looks purely mechanical

nate-chandler committed:
[ABI] Added signedAsData bit to method descriptor flags.

It is necessary to determine whether a given method in a vtable or a
witness table ought to be signed as data or as code.  For example, async
functions pointers must be signed as data.

nate-chandler commented:
@swift-ci please smoke test

nate-chandler commented:
@jckarter Does the new name look good?

nate-chandler commented:
@swift-ci please smoke test macos platform
